tests/unit/customizations/test_overridesslcommonname.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from awscli.testutils import mock, unittest
+from awscli.testutils import BaseAWSCommandParamsTest
+from awscli.customizations.overridesslcommonname import update_endpoint_url
+
+
+class TestOverrideSslCommonName(BaseAWSCommandParamsTest):
+    prefix = 'sqs list-queues'
+
+    def setUp(self):
+        super().setUp()
+        self.parsed_globals = mock.Mock()
+        self.parsed_globals.endpoint_url = None
+        self.parsed_globals.region = None
+        self.parsed_globals.command = 'sqs'
+
+    def test_update_endpoint_url(self):
+        update_endpoint_url(self.driver.session, self.parsed_globals)
+        self.assertEqual(
+            self.parsed_globals.endpoint_url,
+            'https://sqs.us-east-1.amazonaws.com'
+        )
+
+    def test_dont_modify_provided_region_url(self):
+        self.parsed_globals.endpoint_url = 'https://test.com'
+        self.parsed_globals.region = 'us-west-1'
+        update_endpoint_url(self.driver.session, self.parsed_globals)
+        self.assertEqual(self.parsed_globals.endpoint_url, 'https://test.com')
+        self.assertEqual(self.parsed_globals.region, 'us-west-1')
+
+    def test_other_dns_suffixes(self):
+        self.parsed_globals.region = 'cn-north-1'
+        update_endpoint_url(self.driver.session, self.parsed_globals)
+        self.assertEqual(self.parsed_globals.endpoint_url, 'https://sqs.cn-north-1.amazonaws.com.cn')
```

Comment:
This is following the `{service}.{region}.{dnsSuffix}` convention that I thought was supposed to be applied to all endpoints. Is that not the case? As for `sqs` vs. `queue`, I was just grabbing the endpoint prefix from the service model, but since it seems arbitrary for this service, I'll hardcode for each region.